src/main/java/com/redislabs/redisgraph/impl/ResultSetImpl.java

```diff
@@ -1,80 +1,266 @@
 package com.redislabs.redisgraph.impl;
 
+import com.redislabs.redisgraph.Header;
+import com.redislabs.redisgraph.Record;
+import com.redislabs.redisgraph.ResultSet;
+import com.redislabs.redisgraph.Statistics;
+import redis.clients.jedis.util.SafeEncoder;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.NoSuchElementException;
-import java.util.stream.Collectors;
+import java.util.Objects;
 
-import com.redislabs.redisgraph.Record;
-import com.redislabs.redisgraph.ResultSet;
-import com.redislabs.redisgraph.Statistics;
+public class ResultSetImpl implements ResultSet {
 
-import redis.clients.jedis.util.SafeEncoder;
+    private Header header = new HeaderImpl(new ArrayList<>());
```

Comment:
why do you init it here? It seems like a waste overriden by           
```
  parseHeader((List<List<Object>>) rawResponse.get(0));
```

Same about statistics and results

Reply:
a resultset is a complete object.  I don't want to return null at getHeader(), so i return an empty header

Reply:
fixed